src/frontends/src/query_frontend.hpp

```diff
@@ -287,6 +289,18 @@ class RequestHandler {
             request->content.string(), name, versioned_models, policy,
             latency_slo_micros, input_type);
         prediction.then([response, app_metrics](boost::future<Response> f) {
+          if (f.has_exception()) {
+            try {
+              boost::rethrow_exception(f.get_exception_ptr());
```

Comment:
Why are we re-throwing the exception here? This will immediately be caught by the catch statement right?